sysdeps/linux/generic/sysdeps.cpp

```diff
@@ -1579,7 +1579,10 @@ int sys_waitid(idtype_t idtype, id_t id, siginfo_t *info, int options) {
 	return sc_int_result<int>(ret);
 }
 
-int sys_clock_set(int clock, const struct timespec *tp) {
+int sys_clock_set(int clock, time_t secs, long nanos) {
+	struct timespec *tp = {};
+	tp->tv_sec = secs;
+	tp->tv_nsec = nanos;
```

Comment:
This is a nullptr dereference, isn't it?
```suggestion
	struct timespec tp{};
	tp.tv_sec = secs;
	tp.tv_nsec = nanos;
```

Reply:
Doing that breaks the GHA task on compiling linux sysdeps
```c
In file included from ../sysdeps/linux/generic/sysdeps.cpp:21:
../sysdeps/linux/generic/cxx-syscall.hpp: In instantiation of 'mlibc::sc_result_t mlibc::do_syscall(int, T ...) [with T = {int, timespec}]':
../sysdeps/linux/generic/sysdeps.cpp:1593:23:   required from here
 1593 |         auto ret = do_syscall(SYS_clock_settime, clock, tp);
      |                    ~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../sysdeps/linux/generic/cxx-syscall.hpp:80:77: error: no matching function for call to 'sc_cast(timespec&)'
   80 |                 return static_cast<sc_result_t>(do_nargs_syscall(sc, sc_cast(args)...));
      |                                                                      ~~~~~~~^~~~~~
../sysdeps/linux/generic/cxx-syscall.hpp:75:26: note: candidate: 'sc_word_t mlibc::sc_cast(long int)'
   75 |         inline sc_word_t sc_cast(long x) { return x; }
      |                          ^~~~~~~
../sysdeps/linux/generic/cxx-syscall.hpp:75:39: note:   no known conversion for argument 1 from 'timespec' to 'long int'
   75 |         inline sc_word_t sc_cast(long x) { return x; }
      |                                  ~~~~~^
../sysdeps/linux/generic/cxx-syscall.hpp:76:26: note: candidate: 'sc_word_t mlibc::sc_cast(const void*)'
   76 |         inline sc_word_t sc_cast(const void *x) { return reinterpret_cast<sc_word_t>(x); }
      |                          ^~~~~~~
../sysdeps/linux/generic/cxx-syscall.hpp:76:46: note:   no known conversion for argument 1 from 'timespec' to 'const void*'
   76 |         inline sc_word_t sc_cast(const void *x) { return reinterpret_cast<sc_word_t>(x); }
      |                                  ~~~~~~~~~~~~^
      ```